code/modules/mob/mob_helpers.dm

```diff
@@ -256,19 +256,20 @@
 	for(var/datum/multilingual_say_piece/S in message_pieces)
 		S.message = stars(S.message, pr)
 
-/proc/slur(phrase, list/slurletters = ("'"))//use a different list as an input if you want to make robots slur with $#@%! characters
+/proc/slur(phrase, list/slurletters = ("'"), robot = FALSE) //use a different list as an input if you want to make robots slur with $#@%! characters
 	phrase = html_decode(phrase)
 	var/leng=length(phrase)
 	var/counter=length(phrase)
 	var/newphrase=""
 	var/newletter=""
 	while(counter>=1)
 		newletter=copytext(phrase,(leng-counter)+1,(leng-counter)+2)
-		if(rand(1,3)==3)
-			if(lowertext(newletter)=="o")	newletter="u"
-			if(lowertext(newletter)=="s")	newletter="ch"
-			if(lowertext(newletter)=="a")	newletter="ah"
-			if(lowertext(newletter)=="c")	newletter="k"
+		if(!robot)
+			if(rand(1,3)==3)
+				if(lowertext(newletter)=="o")	newletter="u"
+				if(lowertext(newletter)=="s")	newletter="ch"
+				if(lowertext(newletter)=="a")	newletter="ah"
+				if(lowertext(newletter)=="c")	newletter="k"
```

Comment:
```suggestion
   if(!robot)
       while(counter>=1)
		  newletter=copytext(phrase,(leng-counter)+1,(leng-counter)+2)
			if(rand(1,3)==3)
				if(lowertext(newletter)=="o")	newletter="u"
				if(lowertext(newletter)=="s")	newletter="ch"
				if(lowertext(newletter)=="a")	newletter="ah"
				if(lowertext(newletter)=="c")	newletter="k"
```
The check here should be outside the loop, so we can just skip the loop and not be checked multiple times

Reply:
Careful, there's two types of character replacement for each checked, robot only skips the first one.

What you suggested would make robot immediately return the string with no modification.

Reply:
Ah, it is. Whoops. Thought the switch statement was outside of the loop. Best to ignore this then.

Reply:
Should I close this conversation then?

Reply:
This one is good to mark as resolved, check out my other suggestions before you close em though